### ISSUE TITLE
v5.31.4-270-g059703b088 breaks ETHER/Devel-Declare-0.006019.tar.gz 

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#17167]
Migrated from [rt.perl.org#134468](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=134468) (status was 'open')

Searchable as RT134468$

**Comment by p5pRT** [comment]
### From @andk

https://rt.cpan.org/Ticket/Display.html?id=130637

Also affected&#8203;: SREZIC/Tk\-804\.034\.tar\.gz
&nbsp;  https://rt.cpan.org/Ticket/Display.html?id=130638

Thanks\,
\-\- 
andreas
PS&#8203;: perl\,perl\,perl

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Oct 02\, 2019 at 11&#8203;:51&#8203;:04PM \-0700\, \(Andreas J\. Koenig\) \(via RT\) wrote&#8203;:
> \# New Ticket Created by  \(Andreas J\. Koenig\) 
> \# Please include the string&#8203;:  \[perl \#134468\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=134468 >
> 
> 
> https://rt.cpan.org/Ticket/Display.html?id=130637
> 
> Also affected&#8203;: SREZIC/Tk\-804\.034\.tar\.gz
>   https://rt.cpan.org/Ticket/Display.html?id=130638

That's&#8203;:

commit 059703b088f44d5665f67fba0b9d80cad89085fd
Author&#8203;:     Karl Williamson \<khw@&#8203;cpan\.org>
AuthorDate&#8203;: Sun Sep 29 15&#8203;:16&#8203;:24 2019 \-0600
Commit&#8203;:     Karl Williamson \<khw@&#8203;cpan\.org>
CommitDate&#8203;: Sun Sep 29 16&#8203;:15&#8203;:04 2019 \-0600

&nbsp;    Remove deprecated character classification/case changing macros
&nbsp;    
&nbsp;    It has been deprecated since 5\.26 to use various macros that deal with
&nbsp;    UTF\-8 inputs but don't have a parameter indicating the maximum length
&nbsp;    beyond which we should not look\.  This commit changes all such macros\,
&nbsp;    as threatened in existing documentation and warning messages\, to have an
&nbsp;    extra parameter giving the length\.
&nbsp;    
&nbsp;    This was originally scheduled to happen in 5\.30\, but was delayed because
&nbsp;    it broke some CPAN modules\, and there wasn't really a good way around
&nbsp;    it\.  But now that Devel&#8203;::PPPort 3\.54 is out\, ppport\.h has new facilities
&nbsp;    for getting modules making these changes to work with older Perl
&nbsp;    releases\.


\-\- 
No matter how many dust sheets you use\, you will get paint on the carpet\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu\, 03 Oct 2019 06&#8203;:51&#8203;:03 GMT\, andreas\.koenig\.7os6VVqR@&#8203;franz\.ak\.mind\.de wrote&#8203;:
> https://rt.cpan.org/Ticket/Display.html?id=130637
> 
> Also affected&#8203;: SREZIC/Tk\-804\.034\.tar\.gz
>   https://rt.cpan.org/Ticket/Display.html?id=130638
> 
> Thanks\,

In the case of Devel\-Declare\, it's a build\-time failure involving a file ominously called "stolen\_chunk\_of\_toke\.c"\.  See attachment wherein I installed perl 5 blead at HEAD and then attempted to install Devel&#8203;::Declare\.

In the case of Tk\, while we get 142 build\-time warnings\, 'make' completes and the failures are in 'make test'\.  See attachment\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @jkeenan

"/home/jkeenan/testing/blead/bin/perl" \-MExtUtils&#8203;::Command&#8203;::MM \-e 'cp\_nonempty' \-\- Declare\.bs blib/arch/auto/Devel/Declare/Declare\.bs 644
cc \-c  \-I/home/jkeenan/testing/blead/lib/perl5/site\_perl/5\.31\.5/x86\_64\-linux/B/Hooks/OP/Check/Install \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-O2   \-DVERSION=\\"0\.006019\\" \-DXS\_VERSION=\\"0\.006019\\" \-fPIC "\-I/home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/CORE"   Declare\.c
In file included from /home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/CORE/regexp\.h&#8203;:21&#8203;:0\,
&nbsp;                 from /home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/CORE/perl\.h&#8203;:3830\,
&nbsp;                 from Declare\.xs&#8203;:3&#8203;:
stolen\_chunk\_of\_toke\.c&#8203;: In function ‘S\_scan\_word’&#8203;:
/home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/CORE/utf8\.h&#8203;:643&#8203;:13&#8203;: warning&#8203;: implicit declaration of function ‘\_is\_utf8\_FOO’; did you mean ‘\_is\_uni\_FOO’? \[\-Wimplicit\-function\-declaration\]
&nbsp;             \_is\_utf8\_FOO\(\_CC\_IDFIRST\, \(const U8 \*\) p\, "isIDFIRST\_lazy\_if"\,  \\
&nbsp;             ^
/home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/CORE/utf8\.h&#8203;:643&#8203;:13&#8203;: note&#8203;: in definition of macro ‘isIDFIRST\_lazy\_if’
&nbsp;             \_is\_utf8\_FOO\(\_CC\_IDFIRST\, \(const U8 \*\) p\, "isIDFIRST\_lazy\_if"\,  \\
&nbsp;             ^~~~~~~~~~~~
In file included from Declare\.xs&#8203;:7&#8203;:0&#8203;:
stolen\_chunk\_of\_toke\.c&#8203;:449&#8203;:58&#8203;: error&#8203;: macro "isALNUM\_utf8" requires 2 arguments\, but only 1 given
&nbsp;  else if \(UTF && UTF8\_IS\_START\(\*s\) && isALNUM\_utf8\(\(U8\*\)s\)\) \{
&nbsp;                                                          ^
stolen\_chunk\_of\_toke\.c&#8203;:449&#8203;:39&#8203;: error&#8203;: ‘isALNUM\_utf8’ undeclared \(first use in this function\)
&nbsp;  else if \(UTF && UTF8\_IS\_START\(\*s\) && isALNUM\_utf8\(\(U8\*\)s\)\) \{
&nbsp;                                       ^~~~~~~~~~~~
stolen\_chunk\_of\_toke\.c&#8203;:449&#8203;:39&#8203;: note&#8203;: each undeclared identifier is reported only once for each function it appears in
stolen\_chunk\_of\_toke\.c&#8203;: In function ‘S\_scan\_ident’&#8203;:
stolen\_chunk\_of\_toke\.c&#8203;:935&#8203;:62&#8203;: error&#8203;: macro "isALNUM\_utf8" requires 2 arguments\, but only 1 given
&nbsp;      else if \(UTF && UTF8\_IS\_START\(\*s\) && isALNUM\_utf8\(\(U8\*\)s\)\) \{
&nbsp;                                                              ^
stolen\_chunk\_of\_toke\.c&#8203;:935&#8203;:43&#8203;: error&#8203;: ‘isALNUM\_utf8’ undeclared \(first use in this function\)
&nbsp;      else if \(UTF && UTF8\_IS\_START\(\*s\) && isALNUM\_utf8\(\(U8\*\)s\)\) \{
&nbsp;                                           ^~~~~~~~~~~~
Makefile&#8203;:342&#8203;: recipe for target 'Declare\.o' failed
make&#8203;: \*\*\* \[Declare\.o\] Error 1

**Comment by p5pRT** [comment]
### From @jkeenan

[134468.Tk.build.log.gz](https://rt-archive.perl.org/perl5/Ticket/Attachment/1656575/843535/134468.Tk.build.log.gz)

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu\, 03 Oct 2019 12&#8203;:45&#8203;:31 GMT\, jkeenan wrote&#8203;:
> On Thu\, 03 Oct 2019 06&#8203;:51&#8203;:03 GMT\,
> andreas\.koenig\.7os6VVqR@&#8203;franz\.ak\.mind\.de wrote&#8203;:
> > https://rt.cpan.org/Ticket/Display.html?id=130637
> >
> > Also affected&#8203;: SREZIC/Tk\-804\.034\.tar\.gz
> >   https://rt.cpan.org/Ticket/Display.html?id=130638
> >
> > Thanks\,
> 
> In the case of Devel\-Declare\, it's a build\-time failure involving a
> file ominously called "stolen\_chunk\_of\_toke\.c"\.  See attachment
> wherein I installed perl 5 blead at HEAD and then attempted to install
> Devel&#8203;::Declare\.
> 
> In the case of Tk\, while we get 142 build\-time warnings\, 'make'
> completes and the failures are in 'make test'\.  See attachment\.

Tk test failures are associated with "undefined symbol"\, either 'Perl\_to\_utf8\_lower' or 'Perl\_to\_utf8\_upper'\.  See attachment\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @jkeenan

\[Tk\-804\.034\] 520 $ bleadprove \-vb t/regexp\.t 
t/regexp\.t \.\. 
1\.\.22
ok 1 \- Contents as expected
ok 2 \- Correct \-exact postion
ok 3 \- Correct \-exact length
ok 4 \- Correct \-regexp postion
ok 5 \- Correct \-regexp length
ok 6 \- Correct non\-match
ok 7 \- Correct \-regexp \-nocase
/home/jkeenan/testing/blead/bin/perl&#8203;: symbol lookup error&#8203;: /home/jkeenan/\.cpanm/work/1570105828\.4874/Tk\-804\.034/blib/arch/auto/Tk/Tk\.so&#8203;: undefined symbol&#8203;: Perl\_to\_utf8\_lower
Dubious\, test returned 127 \(wstat 32512\, 0x7f00\)
Failed 15/22 subtests 

Test Summary Report


---
t/regexp\.t \(Wstat&#8203;: 32512 Tests&#8203;: 7 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 127
&nbsp;  Parse errors&#8203;: Bad plan\.  You planned 22 tests but ran 7\.
Files=1\, Tests=7\,  0 wallclock secs \( 0\.02 usr  0\.00 sys \+  0\.09 cusr  0\.00 csys =  0\.11 CPU\)
Result&#8203;: FAIL
\[Tk\-804\.034\] 521 $ bleadprove \-vb t/Require\.t 
t/Require\.t \.\. Can't load '/home/jkeenan/\.cpanm/work/1570105828\.4874/Tk\-804\.034/blib/arch/auto/Tk/Tk\.so' for module Tk&#8203;: /home/jkeenan/\.cpanm/work/1570105828\.4874/Tk\-804\.034/blib/arch/auto/Tk/Tk\.so&#8203;: undefined symbol&#8203;: Perl\_to\_utf8\_upper at /home/jkeenan/testing/blead/lib/perl5/5\.31\.5/x86\_64\-linux/DynaLoader\.pm line 193\.
 at t/Require\.t line 5\.
Compilation failed in require at t/Require\.t line 5\.
Dubious\, test returned 2 \(wstat 512\, 0x200\)
No subtests run 

Test Summary Report


---
t/Require\.t \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;  Non\-zero exit status&#8203;: 2
&nbsp;  Parse errors&#8203;: No plan found in TAP output
Files=1\, Tests=0\,  0 wallclock secs \( 0\.02 usr  0\.00 sys \+  0\.02 cusr  0\.00 csys =  0\.04 CPU\)
Result&#8203;: FAIL
\[Tk\-804\.034\] 522 $ bleadprove \-vb t/text\.t
t/text\.t \.\. 
1\.\.415
ok 1 \- use Tk&#8203;::Text;
\# snip
ok 303 \- TextSearchCmd procedure\, \-count option
ok 304
ok 305
/home/jkeenan/testing/blead/bin/perl&#8203;: symbol lookup error&#8203;: /home/jkeenan/\.cpanm/work/1570105828\.4874/Tk\-804\.034/blib/arch/auto/Tk/Tk\.so&#8203;: undefined symbol&#8203;: Perl\_to\_utf8\_lower
Dubious\, test returned 127 \(wstat 32512\, 0x7f00\)
Failed 110/415 subtests 
&nbsp;    \(less 21 skipped subtests&#8203;: 284 okay\)
&nbsp;    \(1 TODO test unexpectedly succeeded\)

Test Summary Report


---
t/text\.t \(Wstat&#8203;: 32512 Tests&#8203;: 305 Failed&#8203;: 0\)
&nbsp;  TODO passed&#8203;:   121
&nbsp;  Non\-zero exit status&#8203;: 127
&nbsp;  Parse errors&#8203;: Bad plan\.  You planned 415 tests but ran 305\.
Files=1\, Tests=305\,  1 wallclock secs \( 0\.06 usr  0\.01 sys \+  0\.21 cusr  0\.01 csys =  0\.29 CPU\)
Result&#8203;: FAIL

**Comment by karenetheridge** [comment]
This should be fixed with https://metacpan.org/release/ETHER/Devel-Declare-0.006_020.
